src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -156,3 +145,51 @@ def create_rados_config_obj(self,
                 # Create an empty config object
                 logger.info('Creating rados config object: %s' % obj)
                 ioctx.write_full(obj, ''.encode('utf-8'))
+
+    def create_keyring(self, daemon_spec: CephadmDaemonSpec[NFSServiceSpec]) -> str:
+        assert daemon_spec.spec
+        daemon_id = daemon_spec.daemon_id
+        spec = daemon_spec.spec
+        entity: AuthEntity = self.get_auth_entity(daemon_id)
+
+        osd_caps = 'allow rw pool=%s' % (spec.pool)
+        if spec.namespace:
+            osd_caps = '%s namespace=%s' % (osd_caps, spec.namespace)
+
+        logger.info('Create keyring: %s' % entity)
+        ret, keyring, err = self.mgr.check_mon_command({
+            'prefix': 'auth get-or-create',
+            'entity': entity,
+            'caps': ['mon', 'allow r',
+                     'osd', osd_caps],
+        })
+
+        return keyring
+
+    def create_rgw_keyring(self, daemon_spec: CephadmDaemonSpec[NFSServiceSpec]) -> str:
+        daemon_id = daemon_spec.daemon_id
+        entity: AuthEntity = self.get_auth_entity(f'{daemon_id}-rgw')
+
+        logger.info('Create keyring: %s' % entity)
+        ret, keyring, err = self.mgr.check_mon_command({
+            'prefix': 'auth get-or-create',
+            'entity': entity,
+            'caps': ['mon', 'allow r',
+                     'osd', 'allow rwx'],
```

Comment:
Why does NFS-Ganesha need access to the OSDs when talking to RGW?

Reply:
Do you have a more restrictive set of perms in mind?

Reply:
I think it just needs `mon "allow r"`? cc @cbodley @mattbenjamin

Reply:
librgw is a librados client. does that mean it needs osd caps?

Reply:
I thought librgw was just talking to the monitors and not the OSDs but maybe I misunderstood.

In any case, it should not have unlimited access to the OSDs with `osd "allow rwx"`.

Reply:
librgw, when initialized, starts an entire radosgw absent only the HTTP front-ends and (by default) some of the background services such as lifecycle.  By design, it accesses services in the Ceph cluster the same as an ordinary radosgw would.  It needs the same permissions to execute--and even identifies itself with the same client string.

Reply:
I see. Thanks for explaining @mattbenjamin. How do you normally restrict rgw instances though in terms of osd caps?

Reply:
@batrick it is possible to restrict rados caps to just the specific pools that a radosgw cluster needs for index, data, log, and data-extra.  The most complete example I know of is by @leseb recently in setting up rook external clusters.  We could probably re-use that here.